testing/run_tests.py

```diff
@@ -75,19 +75,23 @@ def run_cmd(
   print(f'Running command "{command_string}"')
 
   start_time = time.time()
-  collect_output = forbidden_output or allowed_failure_output
-  stdout_pipe = sys.stdout if not collect_output else subprocess.PIPE
-  stderr_pipe = sys.stderr if not collect_output else subprocess.PIPE
 
   process = subprocess.Popen(
       cmd,
-      stdout=stdout_pipe,
-      stderr=stderr_pipe,
+      stdout=subprocess.PIPE,
+      stderr=subprocess.STDOUT,
       env=env,
       universal_newlines=True,
       **kwargs
   )
-  stdout, stderr = process.communicate()
+  output = ''
+
+  for line in iter(process.stdout.readline, ''):
+    output += line
+    sys.stdout.write(line)
+
+  sys.stdout.flush()
+  process.wait()
```

Comment:
Does `process.stdout.readline` drain the buffer to avoid the deadlock concern with `wait()` called out in the docs? https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait

Reply:
That's my understanding.  It's reading the pipe until EOF so there should be no more waiting on the pipe by the time we call `wait`.